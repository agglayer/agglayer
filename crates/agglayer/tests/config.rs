use assert_cmd::Command;

#[test]
fn config_display() -> Result<(), Box<dyn std::error::Error>> {
    let mut cmd = Command::cargo_bin("agglayer")?;
    cmd.args(["config", "--base-dir", "/tmp/agglayer-test"]);

    let output = cmd.assert().success();

    let result: &str = std::str::from_utf8(&output.get_output().stdout)?;

    insta::assert_snapshot!(sanitize_config_folder_path(result));

    Ok(())
}

#[test]
fn prover_config_display() -> Result<(), Box<dyn std::error::Error>> {
    let mut cmd = Command::cargo_bin("agglayer")?;
    cmd.args(["prover-config"]);

    let output = cmd.assert().success();

    let result: &str = std::str::from_utf8(&output.get_output().stdout)?;

    insta::assert_snapshot!(sanitize_config_folder_path(result));

    Ok(())
}

pub fn sanitize_config_folder_path(cmd_out: &str) -> String {
    let dir = dirs::config_dir().unwrap().join("agglayer");
    cmd_out.replace(
        dir.to_str().expect("Unable to transform config dir"),
        "/tmp/agglayer-test",
    )
}
