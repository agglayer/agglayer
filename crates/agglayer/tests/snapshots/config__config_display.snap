---
source: crates/agglayer/tests/config.rs
assertion_line: 12
expression: result
---
[full-node-rpcs]

[proof-signers]

[log]
level = "info"
outputs = []
format = "pretty"

[rpc]
port = 9090
host = "0.0.0.0"

[outbound.rpc.settle]
max-retries = 3
retry-interval = 7
confirmations = 1

[l1]
chain-id = 1337
node-url = "http://zkevm-mock-l1-network:8545/"
rollup-manager-contract = "0xb7f8bc63bbcad18155201308c8f3540b07f84f5e"

[auth.local]
private-keys = []

[telemetry]
prometheus-addr = "0.0.0.0:3000"

[epoch.time-clock]
epoch-duration = 5

[shutdown]
runtime-timeout = 5

[certificate-orchestrator]
input-backpressure-buffer-size = 1000

[certificate-orchestrator.prover.sp1-local]
